pom.xml

```diff
@@ -174,8 +184,9 @@
          If you update a dependendy version so it is no longer a SNAPSHOT
          please update the snapshot-shims profile as well so it is accurate -->
         <spark301.version>3.0.1</spark301.version>
+        <spark301db.version>3.0.1-databricks</spark301db.version>
         <spark302.version>3.0.2-SNAPSHOT</spark302.version>
-        <spark311.version>3.1.1-SNAPSHOT</spark311.version>
+        <spark311.version>3.1.1</spark311.version>
```

Comment:
This PR is suddenly doing something much more than just fixing an NPE.  It's updating the build to pull 3.1.1 instead of a snapshot, arguably a separate PR.

Reply:
surely it can be a separate PR, but it was definitely a prerequisite for passing the test locally

Reply:
No doubt.  However this PR is now submarining a much more impactful change than the headline implies.   One of my pet peeves is PRs with a headline like "fix some minor test thing" that ends up changing way more than that.